### PR TITLE
fix(DTFS2-7159): fix ts-node type checking in all services

### PR DESCRIPTION
**Pull request by JoshBinns2000**
## Introduction :pencil2:
The `ts-node` type checking rules differ slightly from the standard typescript compiler. This can lead to situations where `tsc --noEmit` will pass with no errors, but the container will fail to run due to a typescript type-checking error.

## Resolution :heavy_check_mark:
After some reading, a simple solution to this is to include the below option in `tsconfig.json`. This tells `ts-node` to not use its default `include`/`exclude` options when type checking, but instead use the `tsconfig.json` `include`/`exclude` options.

```json
{
  "ts-node": {
    "files": true
  },
  "include": [...],
  "exclude": [...]
}
```

For further reading, refer to the below links:
- https://stackoverflow.com/questions/67436289/typescripts-declaration-merging-not-working-as-expected-using-ts-node
- https://github.com/TypeStrong/ts-node/issues/1132#issuecomment-716642560

### CODE REVIEW

File: e2e-tests/tsconfig.json [hunk 1]
```diff
@@ -98,6 +98,9 @@
       // "skipDefaultLibCheck": true,                      /* Skip type checking .d.ts files that are included with TypeScript. */
       "skipLibCheck": true                                 /* Skip type checking all .d.ts files. */
     },
+    "ts-node": {
+      "files": true
+    },
     "include": ["**/*.ts"],
     "exclude": ["./node_modules/*", "node_modules"]
   }
```
Comment (MarRobSoftwire) [inline e2e-tests/tsconfig.json:106]: legacy but fwiw no new line at end of file

File: libs/common/tsconfig.json [hunk 2]
```diff
@@ -28,5 +28,9 @@
     /* Completeness */
     "skipLibCheck": true
   },
-  "include": ["**/*.ts"]
+  "ts-node": {
+    "files": true
+  },
+  "include": ["**/*.ts"],
+  "exclude": []
```
Comment (MarRobSoftwire) [inline libs/common/tsconfig.json:35]: Do we want to explicitly ignore `node_modules` as in other places?

File: e2e-tests/tsconfig.json (hunk 1 above)
Comment (MarRobSoftwire) [reply e2e-tests/tsconfig.json:106]: Also in a couple other places

**Review by MarRobSoftwire — DISMISSED** [review]

File: libs/common/tsconfig.json (hunk 2 above)
Comment (JoshBinns2000) [reply libs/common/tsconfig.json:35]: I'm not sure...I think for consistency we probably should, so I'll add that change and see if everything still works

**Review by MarRobSoftwire — APPROVED** [review]

**Pull request merged**